src/chunk_index.c

```diff
@@ -606,7 +660,12 @@ chunk_index_tuple_delete(TupleInfo *ti, void *data)
 		};
 
 		if (OidIsValid(idxobj.objectId))
-			performDeletion(&idxobj, DROP_RESTRICT, 0);
+		{
+			ObjectAddresses *objects = new_object_addresses();
+			chunk_collect_objects_for_deletion(&idxobj, objects);
```

Comment:
Could you comment on the difference between collecting the dependencies and dropping them explicitly here versus
`` performDeletion(&idxobj, DROP_CASCCADE,...); ```

Reply:
Done.